src/Http/Http/perf/SendFileBenchmark.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Http.Extensions
+{
+    public class SendFileBenchmark
+    {
+        [Params(512,
+            1024,
+            1024 * 2,
+            1024 * 4,
+            1024 * 8,
+            1024 * 16,
+            1024 * 32,
+            1024 * 64,
+            1024 * 128,
+            1024 * 256,
+            1024 * 512,
+            1024 * 1024,
+            1024 * 1024 * 2,
+            1024 * 1024 * 4,
+            1024 * 1024 * 8,
+            1024 * 1024 * 16,
+            1024 * 1024 * 32,
+            1024 * 1024 * 64)]
+        public int FileSize { get; set; }
+
+        private string filePath;
+
+        [GlobalSetup]
+        public async Task Setup()
+        {
+            Random random = new Random();
+            var content = new byte[FileSize];
+            random.NextBytes(content);
+
+            filePath = $"{Path.GetTempPath()}Benchmark TempFile - {FileSize}B.txt";
+
+            await File.WriteAllBytesAsync(filePath, content);
+        }
+
+        [Benchmark]
+        public async Task Pipe()
+        {
+            var memorySteam = new MemoryStream();
+            var pipeWriter = PipeWriter.Create(memorySteam);
```

Comment:
Creating the Pipe over a MemoryStream isn't representative of a real Pipe. Maybe create a full Pipe and disable the buffer limits?

Reply:
Yea it occurs to me that these microbenchmarks won't really show the differences we're trying to measure.